repository/src/main/java/org/apache/atlas/repository/store/graph/v2/AtlasRelationshipStoreV2.java

```diff
@@ -936,18 +936,42 @@ private void addRelationshipMetadataForNotificationEvent(Set<AtlasRelationship>
     }
 
     private static Map<String, String> builsESDocIdMapping(AtlasRelationship r) {
+
+        final Map<String, String> esDocIdMapping = new HashMap<>();
+
+        if(r == null || r.getEnd1() == null || r.getEnd2() == null) {
+            LOG.warn(" AtlasRelationship is null {} " , r);
+            return esDocIdMapping;
+        }
+
         final Map<AtlasObjectId, Object> relationshipEndToVertexIdMap = RequestContext.get().getRelationshipEndToVertexIdMap();
         String end1DocId = "";
         String end2DocId = "";
-        for (AtlasObjectId atlasObjectId : relationshipEndToVertexIdMap.keySet()) {
-            final String docId = JanusUtils.toLongEncoding(relationshipEndToVertexIdMap.get(atlasObjectId));
-            if (atlasObjectId.getGuid().equals(r.getEnd1().getGuid())) {
-                end1DocId = docId;
-            } else if (atlasObjectId.getGuid().equals(r.getEnd2().getGuid())) {
-                end2DocId = docId;
+
+            for (AtlasObjectId atlasObjectId : relationshipEndToVertexIdMap.keySet()) {
+                    if(atlasObjectId == null) {
+                        LOG.warn(" atlasObjectId is null.");
+                        return esDocIdMapping;
+                    }
+
+                    final String docId = JanusUtils.toLongEncoding(relationshipEndToVertexIdMap.get(atlasObjectId));
+                    String guid = atlasObjectId.getGuid();
+                    AtlasObjectId end1 = r.getEnd1();
+                    AtlasObjectId end2 = r.getEnd2();
+
+                    if(guid == null) {
+                        LOG.warn(" atlasObjectId.getGuid() is null. atlasObjectId {}" , atlasObjectId);
+                        return esDocIdMapping;
+                    }
+
+                    if (guid.equals(end1.getGuid())) {
```

Comment:
what if end1 or end2 is null?

Reply:
We are checking this in line 942

Reply:
<img width="706" alt="image" src="https://user-images.githubusercontent.com/59254445/231153747-edb227f1-82df-4a14-b5d5-c22b6bbf9454.png">